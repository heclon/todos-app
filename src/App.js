import React, { Component } from "react";
import { Container, Row, Col, Toast, ToastBody, ToastHeader } from "reactstrap";
import ModalAddEdit from "./Components/Modals/ModalAddEdit";
import DataTable from "./Components/Tables/DataTable";

function countByPriority(items) {
let counterHighPriority = 0;
let counterMediumPriority = 0;
let counterLowPriority = 0;
  for (const item of items) {
    if (item.priorityValue === 1) counterHighPriority++;
    if (item.priorityValue === 2) counterMediumPriority++;
    if (item.priorityValue === 3) counterLowPriority++;
  }
  
return [ counterHighPriority, counterMediumPriority, counterLowPriority ]
}
class App extends Component {
  state = {
    items: [],
    highPriority: 0,
    mediumPriority: 0,
    lowPriority: 0,
  };

  getItems() {
    this.setState({ items: [] });
  }

  addItemToState = (item) => {
    this.setState((prevState) => ({
      items: [...prevState.items, item],
    }));
    this.updateCounters()
  };

  updateState = (item) => {
    const itemIndex = this.state.items.findIndex((data) => data.id === item.id);
    const newArray = [
      ...this.state.items.slice(0, itemIndex),
      item,
      ...this.state.items.slice(itemIndex + 1),
    ];
    this.setState({ items: newArray });
    this.updateCounters()
  };

  deleteItemFromState = (id) => {
    const updatedItems = this.state.items.filter((item) => item.id !== id);
    this.setState({ items: updatedItems });
    this.updateCounters(this.state.items)
  };

  componentDidMount() {
    this.getItems();
  }


  
  updateCounters(){
    const items = this.state.items
    const [ counterHighPriority, 
      counterMediumPriority, 
      counterLowPriority ] =  countByPriority(items)

      
    this.setState({
      highPriority: counterHighPriority,
      mediumPriority: counterMediumPriority,
      lowPriority: counterLowPriority
    })

  }

  render() {

    const [ counterHighPriority, 
      counterMediumPriority, 
      counterLowPriority ] =  countByPriority(this.state.items)

    return (
      <Container className="App">
        <Row>
          <Col>
            <h1 style={{ margin: "20px 0" }}>My To-do's</h1>
          </Col>
        </Row>
        <Row style={{ marginBottom: "3%" }}>
          <Col>
            <ModalAddEdit
              buttonLabel="Add To-do task"
              addItemToState={this.addItemToState}
            />
          </Col>
          <Col>
            <div className="p-3 bg-danger rounded">
              <Toast>
                <ToastHeader>
                  High Priority
                </ToastHeader>
                <ToastBody>
                  {counterHighPriority}
                </ToastBody>
              </Toast>
            </div>
          </Col>
          <Col>
            <div className="p-3 bg-warning rounded">
              <Toast>
                <ToastHeader>
                Medium Priority 
                </ToastHeader>
                <ToastBody>
                {counterMediumPriority}
                </ToastBody>
              </Toast>
            </div>
          </Col>
          <Col>
            <div className="p-3 bg-primary rounded">
              <Toast>
                <ToastHeader>
                  Low Priority
                </ToastHeader>
                <ToastBody>
                {counterLowPriority}
                </ToastBody>
              </Toast>
            </div>
          </Col>
        </Row>
        <Row>
          <Col>
            <DataTable
              items={this.state.items}
              updateState={this.updateState}
              deleteItemFromState={this.deleteItemFromState}
            />
          </Col>
        </Row>
        <Row>
          <Col>
          Created by Hector Longarte</Col>
        </Row>
        <Row>
          <Col>
          Linkedin: </Col>
        </Row>
      </Container>
    );
  }
}

export default App;
