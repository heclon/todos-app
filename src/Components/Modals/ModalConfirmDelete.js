// import React, { Component } from "react";
// import { Button, Modal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap';

// class ModalConfirmDelete extends Component {
//     constructor(props) {
//         super(props);
//         this.state = {
//             modal: false,
//         };
//     }

//     toggle = () => {
//       this.setState((prevState) => ({
//         modal: !prevState.modal,
//       }));
//     };
  
//   render() {
//     const {
//         buttonLabel,
//         className,
//         itemId,
//       } = this.props;

//     return (
//         <div>
//         <Button color="danger" onClick={this.toggle}>{buttonLabel}</Button>
//         <Modal isOpen={this.state.modal} toggle={this.toggle} className={className}>
//             <ModalHeader toggle={this.toggle}>Delete To-do confirmation</ModalHeader>
//             <ModalBody>
//             Are you sure that you wish to delete this task To-do?
//             </ModalBody>
//             <ModalFooter>
//             <Button color="danger" onClick={() => this.props.deleteItemFromState(itemId)}>Yes, delete it</Button>{' '}
//             <Button color="secondary" onClick={this.toggle}>Cancel</Button>
//             </ModalFooter>
//         </Modal>
//         </div>
//     );
//   }
// }

// export default ModalConfirmDelete;