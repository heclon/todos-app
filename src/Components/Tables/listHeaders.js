const listHeaders = [
    {
      Header: "ID",
      className: "t-cell-1 text-left",
      accessor: "id",
      notShowSortingDisplay: true,
    },
    {
      Header: "Task Name",
      accessor: "taskName",
      className: "t-cell-2 text-left",
    },
    {
      Header: "Priority",
      accessor: "priority",
      className: "t-cell-3 text-left",
    },
    {
      Header: "Status",
      accessor: "status",
      className: "t-cell-4 text-center",
    },
    {
      Header: "Actions",
      accessor: "actions",
      className: "t-cell-5 text-center",
    },
  ];

  export default listHeaders;