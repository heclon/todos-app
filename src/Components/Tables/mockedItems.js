const mockedItems = [
  {
    id: "hook",
    taskName: "Demo React Hooks",
    priority: "1-High",
    priorityValue: "1",
    status: "In progress",
    statusValue: "2",
  },
  {
    id: "awsx",
    taskName: "AWS Architecture exams",
    priority: "2-Medium",
    priorityValue: "2",
    status: "Pending",
    statusValue: "1",
  },
  {
    id: "awsd",
    taskName: "AWS Development exam",
    priority: "2-Medium",
    priorityValue: "2",
    status: "In progress",
    statusValue: "2",
  },
  {
    id: "pyad",
    taskName: "Python advanced learning",
    priority: "3-Low",
    priorityValue: "3",
    status: "Pending",
    statusValue: "1",
  },
  {
    id: "zing",
    taskName: "Zing-Zing last",
    priority: "2-Medium",
    priorityValue: "2",
    status: "Done",
    statusValue: "3",
  },
];

export default mockedItems;
