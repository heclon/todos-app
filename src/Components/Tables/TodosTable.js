import React from "react";
import { useTable, useSortBy, usePagination } from "react-table";

import { Table, Container, Row, Col } from "reactstrap";
import { Pagination, PaginationItem, PaginationLink } from "reactstrap";
import classNames from "classnames";
import "./assets/Table.css";
import sortasc from "./assets/sort_asc.png";
import sortdesc from "./assets/sort_desc.png";
import sortboth from "./assets/sort_both.png";
import loaderimage from "./assets/loader-table.gif";
import ModalAddEdit from "../Modals/ModalAddEdit";
import ModalConfirmDelete from "../Modals/ModalConfirmDelete";

const TodosTable = ({
  data,
  columns,
  updateState,
  deleteItemFromState,
  loading = true,
}) => {
  const defaultColumn = React.useMemo(() => ({}), []);

  const {
    getTableProps,
    getTableBodyProps,
    headerGroups,
    page,
    prepareRow,
    canPreviousPage,
    canNextPage,
    pageOptions,
    pageCount,
    gotoPage,
    nextPage,
    previousPage,
    setPageSize,
    state: { pageIndex, pageSize },
  } = useTable(
    {
      columns,
      data,
      initialState: { pageIndex: 0, pageSize: 10 },
      defaultColumn,
    },
    useSortBy,
    usePagination
  );

  return (
    <>
      <Table {...getTableProps()} hover bordered responsive>
        <thead>
          {headerGroups.map((headerGroup) => (
            <>
              <tr
                key={"header"}
                className="theader"
                {...headerGroup.getHeaderGroupProps()}
              >
                {headerGroup.headers.map((column, i) => (
                  <th
                    key={i}
                    {...column.getHeaderProps(column.getSortByToggleProps())}
                  >
                    {column.render("Header")}
                    <span className="float-right">
                      {!column.notShowSortingDisplay ? (
                        column.isSorted ? (
                          column.isSortedDesc ? (
                            <img src={sortdesc} alt="descending" />
                          ) : (
                            <img src={sortasc} alt="ascending" />
                          )
                        ) : (
                          <img src={sortboth} alt="sorting" />
                        )
                      ) : (
                        ""
                      )}
                    </span>
                  </th>
                ))}
              </tr>
            </>
          ))}
        </thead>
        {loading ? (
          <tbody>
            <tr key={0}>
              <td colSpan="10000" className="text-center">
                <img src={loaderimage} alt="Loading..." />
              </td>
            </tr>
          </tbody>
        ) : (
          <>
            {page.length === 0 ? (
              <tbody>
                <tr key={0}>
                  <td colSpan="10000" className="text-left">
                    -- No records found --
                  </td>
                </tr>
              </tbody>
            ) : (
              <tbody {...getTableBodyProps()}>
                {page.map((row, i) => {
                  // console.log(i)
                  prepareRow(row);
                  return (
                    <tr key={row.id} {...row.getRowProps()}>
                      {row.cells.map((cell) => {
                        const header = cell.column.Header;
                        if (header === "Actions") {
                          return (
                            <td
                              {...cell.getCellProps({
                                className: cell.column.className,
                              })}
                            >
                              <Container>
                                <Row>
                                  <Col
                                    style={{ float: "center", margin: "5px" }}
                                  >
                                    <ModalAddEdit
                                      buttonLabel="Edit"
                                      color="warning"
                                      item={data[i]}
                                      updateState={updateState}
                                    />
                                  </Col>
                                  <Col
                                    style={{ float: "center", margin: "5px" }}
                                  >
                                    <ModalConfirmDelete
                                      buttonLabel="Delete"
                                      item={data[i]}
                                      deleteItemFromState={deleteItemFromState}
                                    />
                                  </Col>
                                </Row>
                              </Container>
                            </td>
                          );
                        } else {
                          return (
                            <td
                              {...cell.getCellProps({
                                className: cell.column.className,
                              })}
                            >
                              {cell.render("Cell")}
                            </td>
                          );
                        }
                      })}
                    </tr>
                  );
                })}
              </tbody>
            )}
          </>
        )}
      </Table>

      {page.length > 0 && (
        <div className={classNames("div-pagination", { "d-none": loading })}>
          <div className="div-pagination-2">
            <div className="div-pagination-2-2">
              Showing{" "}
              <select
                className="selectan"
                value={pageSize}
                onChange={(e) => {
                  setPageSize(Number(e.target.value));
                }}
              >
                {[10, 20, 30, 50, 100].map((pageSize) => (
                  <option key={pageSize} value={pageSize}>
                    {pageSize}
                  </option>
                ))}
              </select>{" "}
              record per-page
            </div>
          </div>

          <div className="div-pagination-1">
            Page : {pageIndex + 1} from {pageOptions.length}{" "}
            <Pagination className="pagina">
              <PaginationItem disabled={!canPreviousPage}>
                <PaginationLink onClick={() => gotoPage(0)}>
                  {"<<"}
                </PaginationLink>
              </PaginationItem>
              <PaginationItem disabled={!canPreviousPage}>
                <PaginationLink onClick={() => previousPage()}>
                  {"<"}
                </PaginationLink>
              </PaginationItem>
              <PaginationItem disabled={!canNextPage}>
                <PaginationLink onClick={() => nextPage()}>
                  {">"}
                </PaginationLink>
              </PaginationItem>
              <PaginationItem disabled={!canNextPage}>
                <PaginationLink onClick={() => gotoPage(pageCount - 1)}>
                  {">>"}
                </PaginationLink>
              </PaginationItem>
            </Pagination>
          </div>
        </div>
      )}
    </>
  );
};

export default TodosTable;
