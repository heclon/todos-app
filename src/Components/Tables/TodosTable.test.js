import React from "react";
import { mount } from "enzyme";
import App from "../../App";
import TodosTable from "./TodosTable";
import ModalAddEdit from "../Modals/ModalAddEdit";
import ModalConfirmDelete from "../Modals/ModalConfirmDelete";
import renderer from "react-test-renderer";
import mockedItems from "./mockedItems";
import listHeaders from "./listHeaders";

describe("TodosTable tests", () => {
  let todoTableComponent;
  beforeEach(() => {
    todoTableComponent = mount(
      <TodosTable
        columns={listHeaders}
        data={mockedItems}
        updateState={App.updateState}
        deleteItemFromState={App.deleteItemFromState}
        loading={false}  
    />
    );
  });

  it("snapshot test render correctly TodosTable component", () => {
    const todoTable = renderer
      .create(
        <TodosTable
        columns={listHeaders}
        data={mockedItems}
        updateState={App.updateState}
        deleteItemFromState={App.deleteItemFromState}
        loading={false}  
      />
      )
      .toJSON();
    expect(todoTable).toMatchSnapshot();
  });

  it("Test that TodosTable renders to-do items correctly", () => {
    // there should be one table HTML element rendered
    const table = todoTableComponent.find("table");
    expect(table).toHaveLength(1);

    // that table shoud be rendered with its header
    const headerRow = table.find("thead");
    expect(headerRow).toHaveLength(1);

    // there should be 4 headers
    const headers = headerRow.find("th");
    expect(headers).toHaveLength(5);

    const headerId = headers.at(0);
    expect(headerId.text()).toBe("ID");

    const headerTaskName = headers.at(1);
    expect(headerTaskName.text()).toBe("Task Name");

    const headerPriority = headers.at(2);
    expect(headerPriority.text()).toBe("Priority");

    // there should be 1 table body with 6 rows for mocked items
    const body = table.find("tbody");
    expect(body).toHaveLength(1);

    const rows = body.find("tr");
    expect(rows).toHaveLength(5);

    // there should be a ModalAddEdit for each of the mocked items
    const editButtons = table.find(ModalAddEdit);
    expect(editButtons).toHaveLength(5);

    // and a ModalConfirmDelete for each of the mocked items
    const deleteButtons = table.find(ModalConfirmDelete);
    expect(deleteButtons).toHaveLength(5);

    // now let's test the values passed => the first mocked item vs the first row
    // ID are non-editable and
    // reactstrap displays them as th, not as td
    const tds = body.find("td");
    expect(tds.at(0).text()).toBe(mockedItems[0].id);

    expect(tds.at(1).text()).toBe(mockedItems[0].taskName);
    expect(tds.at(2).text()).toBe(mockedItems[0].priority);
    expect(tds.at(3).text()).toBe(mockedItems[0].status);
  });

  describe("test TodosTable sorts", () => {
    it("Test sort in ASC order upon click on taskName header", () => {
      // Simulate click on Task To-do name
      const thead = todoTableComponent.find("thead");
      const headers = thead.find("th");
      const headerTaskName = headers.at(1);
      headerTaskName.simulate("click");

      // and check for sorted items on the table
      // Now the first item should be:
      // {
      //   id: "awsx",
      //   taskName: "AWS Architecture exams",
      //   priority: "2-Medium",
      //   priorityValue: 2,
      // },
      const tableAfterSort = todoTableComponent.find("table");
      const body = tableAfterSort.find("tbody");
      const tds = body.find("td");

      expect(tds.at(0).text()).toBe("awsx");

      expect(tds.at(1).text()).toBe("AWS Architecture exams");
      expect(tds.at(2).text()).toBe("2-Medium");
      expect(tds.at(3).text()).toBe("Pending");
    });

    it("Test sort in DESC order upon click on task-name header", () => {
      // Simulate click on Task To-do name
      const thead = todoTableComponent.find("thead");
      const headers = thead.find("th");
      const headerTaskName = headers.at(1);

      //To get DESC order we need 2 clicks
      headerTaskName.simulate("click");
      headerTaskName.simulate("click");

      // and check for sorted items on the table
      // Now the first item should be:
      // {
      //   id: "zing",
      //   taskName: "Zing-Zing last",
      //   priority: "2-Medium",
      //   priorityValue: 2,
      // },
      const tableAfterSort = todoTableComponent.find("table");
      const body = tableAfterSort.find("tbody");
      const tds = body.find("td");

      expect(tds.at(0).text()).toBe("zing");
      expect(tds.at(1).text()).toBe("Zing-Zing last");
      expect(tds.at(2).text()).toBe("2-Medium");
      expect(tds.at(3).text()).toBe("Done");
    });

    it("Test sort in ASC order upon click on priority name header", () => {
      // Simulate click on Task To-do name
      const thead = todoTableComponent.find("thead");
      const headers = thead.find("th");
      const headerTaskName = headers.at(2);

      headerTaskName.simulate("click");

      // and check for sorted items on the table
      // Now the first item should be:
      // {
      //   id: "hook",
      //   taskName: "Demo React Hooks",
      //   priority: "1-High",
      //   priorityValue: 1,
      // },
      const tableAfterSort = todoTableComponent.find("table");
      const body = tableAfterSort.find("tbody");
      const tds = body.find("td");

      expect(tds.at(0).text()).toBe("hook");
      expect(tds.at(1).text()).toBe("Demo React Hooks");
      expect(tds.at(2).text()).toBe("1-High");
      expect(tds.at(3).text()).toBe("In progress");
    });

    it("Test sort in DESC order upon click on priority header", () => {
      // Simulate click on Task To-do name
      const thead = todoTableComponent.find("thead");
      const headers = thead.find("th");
      const headerTaskName = headers.at(2);

      headerTaskName.simulate("click");
      headerTaskName.simulate("click");

      // and check for sorted items on the table
      // Now the first item should be:
      // {
      //   id: "pyad",
      //   taskName: "Python advanced learning",
      //   priority: "3-Low",
      //   priorityValue: 3,
      // },
      const tableAfterSort = todoTableComponent.find("table");
      const body = tableAfterSort.find("tbody");
      const tds = body.find("td");

      expect(tds.at(0).text()).toBe("pyad");
      expect(tds.at(1).text()).toBe("Python advanced learning");
      expect(tds.at(2).text()).toBe("3-Low");
      expect(tds.at(3).text()).toBe("Pending");
    });
  });

  describe("Test TodosTable button actions", () => {

    it("Edit button saves the item new state on todos table", () => {

    });
    it("Delete button removes item from todos table", () => {

    });
  });
});
