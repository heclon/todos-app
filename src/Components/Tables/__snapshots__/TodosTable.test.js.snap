// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodosTable tests snapshot test render correctly TodosTable component 1`] = `
Array [
  <div
    className="table-responsive"
  >
    <table
      className="table table-bordered table-hover"
      role="table"
    >
      <thead>
        <tr
          className="theader"
          role="row"
        >
          <th
            colSpan={1}
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "cursor": "pointer",
              }
            }
            title="Toggle SortBy"
          >
            ID
            <span
              className="float-right"
            >
              
            </span>
          </th>
          <th
            colSpan={1}
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "cursor": "pointer",
              }
            }
            title="Toggle SortBy"
          >
            Task Name
            <span
              className="float-right"
            >
              <img
                alt="sorting"
                src=""
              />
            </span>
          </th>
          <th
            colSpan={1}
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "cursor": "pointer",
              }
            }
            title="Toggle SortBy"
          >
            Priority
            <span
              className="float-right"
            >
              <img
                alt="sorting"
                src=""
              />
            </span>
          </th>
          <th
            colSpan={1}
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "cursor": "pointer",
              }
            }
            title="Toggle SortBy"
          >
            Status
            <span
              className="float-right"
            >
              <img
                alt="sorting"
                src=""
              />
            </span>
          </th>
          <th
            colSpan={1}
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "cursor": "pointer",
              }
            }
            title="Toggle SortBy"
          >
            Actions
            <span
              className="float-right"
            >
              <img
                alt="sorting"
                src=""
              />
            </span>
          </th>
        </tr>
      </thead>
      <tbody
        role="rowgroup"
      >
        <tr
          role="row"
        >
          <td
            className="t-cell-1 text-left"
            role="cell"
          >
            hook
          </td>
          <td
            className="t-cell-2 text-left"
            role="cell"
          >
            Demo React Hooks
          </td>
          <td
            className="t-cell-3 text-left"
            role="cell"
          >
            1-High
          </td>
          <td
            className="t-cell-4 text-center"
            role="cell"
          >
            In progress
          </td>
          <td
            className="t-cell-5 text-center"
            role="cell"
          >
            <div
              className="container"
            >
              <div
                className="row"
              >
                <div
                  className="col"
                  style={
                    Object {
                      "float": "center",
                      "margin": "5px",
                    }
                  }
                >
                  <div>
                    <button
                      aria-label={null}
                      className="btn btn-secondary"
                      onClick={[Function]}
                      style={
                        Object {
                          "float": "left",
                          "marginRight": "10px",
                        }
                      }
                      type="button"
                    >
                      Edit
                    </button>
                  </div>
                </div>
                <div
                  className="col"
                  style={
                    Object {
                      "float": "center",
                      "margin": "5px",
                    }
                  }
                >
                  <div>
                    <button
                      aria-label={null}
                      className="btn btn-danger"
                      onClick={[Function]}
                      type="button"
                    >
                      Delete
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </td>
        </tr>
        <tr
          role="row"
        >
          <td
            className="t-cell-1 text-left"
            role="cell"
          >
            awsx
          </td>
          <td
            className="t-cell-2 text-left"
            role="cell"
          >
            AWS Architecture exams
          </td>
          <td
            className="t-cell-3 text-left"
            role="cell"
          >
            2-Medium
          </td>
          <td
            className="t-cell-4 text-center"
            role="cell"
          >
            Pending
          </td>
          <td
            className="t-cell-5 text-center"
            role="cell"
          >
            <div
              className="container"
            >
              <div
                className="row"
              >
                <div
                  className="col"
                  style={
                    Object {
                      "float": "center",
                      "margin": "5px",
                    }
                  }
                >
                  <div>
                    <button
                      aria-label={null}
                      className="btn btn-secondary"
                      onClick={[Function]}
                      style={
                        Object {
                          "float": "left",
                          "marginRight": "10px",
                        }
                      }
                      type="button"
                    >
                      Edit
                    </button>
                  </div>
                </div>
                <div
                  className="col"
                  style={
                    Object {
                      "float": "center",
                      "margin": "5px",
                    }
                  }
                >
                  <div>
                    <button
                      aria-label={null}
                      className="btn btn-danger"
                      onClick={[Function]}
                      type="button"
                    >
                      Delete
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </td>
        </tr>
        <tr
          role="row"
        >
          <td
            className="t-cell-1 text-left"
            role="cell"
          >
            awsd
          </td>
          <td
            className="t-cell-2 text-left"
            role="cell"
          >
            AWS Development exam
          </td>
          <td
            className="t-cell-3 text-left"
            role="cell"
          >
            2-Medium
          </td>
          <td
            className="t-cell-4 text-center"
            role="cell"
          >
            In progress
          </td>
          <td
            className="t-cell-5 text-center"
            role="cell"
          >
            <div
              className="container"
            >
              <div
                className="row"
              >
                <div
                  className="col"
                  style={
                    Object {
                      "float": "center",
                      "margin": "5px",
                    }
                  }
                >
                  <div>
                    <button
                      aria-label={null}
                      className="btn btn-secondary"
                      onClick={[Function]}
                      style={
                        Object {
                          "float": "left",
                          "marginRight": "10px",
                        }
                      }
                      type="button"
                    >
                      Edit
                    </button>
                  </div>
                </div>
                <div
                  className="col"
                  style={
                    Object {
                      "float": "center",
                      "margin": "5px",
                    }
                  }
                >
                  <div>
                    <button
                      aria-label={null}
                      className="btn btn-danger"
                      onClick={[Function]}
                      type="button"
                    >
                      Delete
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </td>
        </tr>
        <tr
          role="row"
        >
          <td
            className="t-cell-1 text-left"
            role="cell"
          >
            pyad
          </td>
          <td
            className="t-cell-2 text-left"
            role="cell"
          >
            Python advanced learning
          </td>
          <td
            className="t-cell-3 text-left"
            role="cell"
          >
            3-Low
          </td>
          <td
            className="t-cell-4 text-center"
            role="cell"
          >
            Pending
          </td>
          <td
            className="t-cell-5 text-center"
            role="cell"
          >
            <div
              className="container"
            >
              <div
                className="row"
              >
                <div
                  className="col"
                  style={
                    Object {
                      "float": "center",
                      "margin": "5px",
                    }
                  }
                >
                  <div>
                    <button
                      aria-label={null}
                      className="btn btn-secondary"
                      onClick={[Function]}
                      style={
                        Object {
                          "float": "left",
                          "marginRight": "10px",
                        }
                      }
                      type="button"
                    >
                      Edit
                    </button>
                  </div>
                </div>
                <div
                  className="col"
                  style={
                    Object {
                      "float": "center",
                      "margin": "5px",
                    }
                  }
                >
                  <div>
                    <button
                      aria-label={null}
                      className="btn btn-danger"
                      onClick={[Function]}
                      type="button"
                    >
                      Delete
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </td>
        </tr>
        <tr
          role="row"
        >
          <td
            className="t-cell-1 text-left"
            role="cell"
          >
            zing
          </td>
          <td
            className="t-cell-2 text-left"
            role="cell"
          >
            Zing-Zing last
          </td>
          <td
            className="t-cell-3 text-left"
            role="cell"
          >
            2-Medium
          </td>
          <td
            className="t-cell-4 text-center"
            role="cell"
          >
            Done
          </td>
          <td
            className="t-cell-5 text-center"
            role="cell"
          >
            <div
              className="container"
            >
              <div
                className="row"
              >
                <div
                  className="col"
                  style={
                    Object {
                      "float": "center",
                      "margin": "5px",
                    }
                  }
                >
                  <div>
                    <button
                      aria-label={null}
                      className="btn btn-secondary"
                      onClick={[Function]}
                      style={
                        Object {
                          "float": "left",
                          "marginRight": "10px",
                        }
                      }
                      type="button"
                    >
                      Edit
                    </button>
                  </div>
                </div>
                <div
                  className="col"
                  style={
                    Object {
                      "float": "center",
                      "margin": "5px",
                    }
                  }
                >
                  <div>
                    <button
                      aria-label={null}
                      className="btn btn-danger"
                      onClick={[Function]}
                      type="button"
                    >
                      Delete
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  <div
    className="div-pagination"
  >
    <div
      className="div-pagination-2"
    >
      <div
        className="div-pagination-2-2"
      >
        Showing
         
        <select
          className="selectan"
          onChange={[Function]}
          value={10}
        >
          <option
            value={10}
          >
            10
          </option>
          <option
            value={20}
          >
            20
          </option>
          <option
            value={30}
          >
            30
          </option>
          <option
            value={50}
          >
            50
          </option>
          <option
            value={100}
          >
            100
          </option>
        </select>
         
        record per-page
      </div>
    </div>
    <div
      className="div-pagination-1"
    >
      Page : 
      1
       from 
      1
       
      <nav
        aria-label="pagination"
        className="pagina"
      >
        <ul
          className="pagination"
        >
          <li
            className="page-item disabled"
          >
            <button
              className="page-link"
              onClick={[Function]}
            >
              &lt;&lt;
            </button>
          </li>
          <li
            className="page-item disabled"
          >
            <button
              className="page-link"
              onClick={[Function]}
            >
              &lt;
            </button>
          </li>
          <li
            className="page-item disabled"
          >
            <button
              className="page-link"
              onClick={[Function]}
            >
              &gt;
            </button>
          </li>
          <li
            className="page-item disabled"
          >
            <button
              className="page-link"
              onClick={[Function]}
            >
              &gt;&gt;
            </button>
          </li>
        </ul>
      </nav>
    </div>
  </div>,
]
`;
